src/python/pants/core/goals/package.py

```diff
@@ -28,11 +28,21 @@ class PackageFieldSet(FieldSet, metaclass=ABCMeta):
     """The fields necessary to build an asset from a target."""
 
 
+@dataclass(frozen=True)
+class BuiltPackageArtifact:
+    """Information about artifacts in a built package.
+
+    Used for logging information about the artifacts that are dumped to the distdir.
+    """
+
+    relpath: Optional[str]
+    extra_log_lines: Tuple[str, ...] = tuple()
+
+
 @dataclass(frozen=True)
 class BuiltPackage:
     digest: Digest
-    relpath: str
-    extra_log_info: Optional[str] = None
+    artifacts: Tuple[BuiltPackageArtifact, ...]
```

Comment:
`pants.util.collections.assert_single_element()` is a bit wordy, but can make it a little more explicit if we expect there to be only one artifact. Something like:
```python
        @property
        def single_artifact(self) -> BuiltPackageArtifact:
            return assert_single_element(self.artifacts)
```